### PR TITLE
fix matchPatter helper code structures

### PR DESCRIPTION
**Pull request by kirylbehan**
I always check source code of the package before using it.
Some problematic structures rushed in, here's a fix if you don't mind .
Thanks

### CODE REVIEW

File: src/internals/helpers.ts [hunk 1]
```diff
@@ -90,9 +89,7 @@ export const matchPattern = (
           endPatterns.every((subPattern, i) =>
             matchPattern(subPattern, endValues[i], select)
           ) &&
-          (variadicPatterns.length === 0
-            ? true
-            : matchPattern(variadicPatterns[0], middleValues, select))
+          matchPattern(variadicPatterns[0], middleValues, select)
```
Comment (gvergnaud) [inline src/internals/helpers.ts:92]: This is incorrect and would likely break test on tuple pattern.

**Review by gvergnaud — COMMENTED** [review]
Hey, I don't accept stylistic code contributions because everyone has different preferences and there is no way any code can esthetically please everyone. I don't think it's worth it. 

If there are specific behaviors you think are bugs and would like to fix, please open an issue first

File: src/internals/helpers.ts (hunk 1 above)
Comment (kirylbehan) [reply src/internals/helpers.ts:92]: above check `if (variadicPatterns.length) {`

**Comment by kirylbehan** [comment]
Its less code, not esthetically better but clear and modern
As you please

**Pull request closed without merging**